accuracy_metric.py

```diff
@@ -0,0 +1,160 @@
+"""
+Automatic accuracy metric for MT-GenEval paper.
+We use the correct (gender-matched) references and incorrect (counterfactual/swapped) references to evaluate accuracy.
+We first check the references for unique/gender-specific terms, then check for overlap with the incorrect gender
+and the hypothesis. If there is overlap, we mark the segment as incorrect.
+"""
+import os
+import string
+import argparse
+
+
+STRIP_PUNCT = str.maketrans(string.punctuation, ' '*len(string.punctuation))
+
+
+def clean_line(line):
+    """
+    For an input line, lowercase it, strip extra spaces, and replace punctuation with spaces.
+
+    :param line: Line to clean.
+    :return: Cleaned version of the line
+    """
+    return line.lower().translate(STRIP_PUNCT).strip()
+
+
+def get_words(line):
+    """
+    Helper function to get the set of words in a line.
+
+    :param line: Line from which to get the words.
+    :return: Set of words in the line.
+    """
+    return set(line.strip().split())
+
+
+def get_trg_correct_incorrect(trg_line, cor_line, inc_line):
+    """
+    Compute overlap between references and translation
+    We first get unique words in each of the references w.r.t each other then we compute their overlap with target
+    """ 
+    # get words for each segment
+    trg_words, cor_words, inc_words = get_words(trg_line), get_words(cor_line), get_words(inc_line)
+    # get unique words in each of the references
+    cor_unique = cor_words - inc_words
+    inc_unique = inc_words - cor_words
+    # now check the words in the target sentence for overlap with incorrect unique words
+    trg_correct = trg_words & cor_unique 
+    trg_incorrect = trg_words & inc_unique
+    return trg_correct, trg_incorrect 
+
+
+def gender_decision(trg_line, cor_line, inc_line):
+    """
+    Check if gender of a sentence is correct based on corresponding correct and incorrect references.
+    Algorithm: We make decision based on whether hyp overlaps with original ref and counterfactual ref
+
+    :param trg_line: Sentence from translation output for which to check gender.
+    :param cor_line: Feminine reference.
```

Comment:
nit: `cor_line` isn't always the feminine reference, right? It's the reference corresponding to the gender based on the source? (also below with `inc_line`)

Reply:
I see below you call them `original ref` and `counterfactual ref`, that might be clearer than `cor` and `inc`.